squidb/src/com/yahoo/squidb/sql/DBObject.java

```diff
@@ -75,27 +75,35 @@ public boolean equals(Object o) {
         if (alias != null ? !alias.equals(dbObject.alias) : dbObject.alias != null) {
             return false;
         }
-        String myExpression = getExpression();
-        String otherExpression = dbObject.getExpression();
-        if (myExpression != null ? !myExpression.equals(otherExpression) : otherExpression != null) {
+        if (expressionForComparison() != null ? !expressionForComparison().equals(dbObject.expressionForComparison())
```

Comment:
I generally like to do the method call once, but not opposed to doing it this way.